server/client.go

```diff
@@ -490,6 +496,18 @@ func (c *client) authViolation() {
 	c.closeConnection()
 }
 
+func (c *client) maxConnLimit() {
+	if c.srv != nil && c.srv.opts.Users != nil {
```

Comment:
Since its global scope I think we can just use the non-user generic one below, don't need the user specific one IMO.